src/lib/schematics/ng-add/theming/theming.ts

```diff
@@ -135,25 +141,31 @@ function addThemeStyleToTarget(project: WorkspaceProject, targetName: string, ho
   host.overwrite('angular.json', JSON.stringify(workspace, null, 2));
 }
 
-/** Throws if the project is not using the default Angular devkit builders. */
-function assertDefaultBuildersConfigured(project: WorkspaceProject) {
-  checkProjectTargetBuilder(project, 'build', '@angular-devkit/build-angular:browser');
-  checkProjectTargetBuilder(project, 'test', '@angular-devkit/build-angular:karma');
-}
-
 /**
- * Checks if the specified project target is configured with the default builders which are
- * provided by the Angular CLI.
+ * Validates that the specified project target is configured with the default builders which are
+ * provided by the Angular CLI. If the configured builder does not match the default builder,
+ * this function can either throw or just show a warning.
  */
-function checkProjectTargetBuilder(project: WorkspaceProject, targetName: string,
-                            defaultBuilder: string) {
-
+function validateDefaultTargetBuilder(project: WorkspaceProject, targetName: 'build' | 'test') {
+  const defaultBuilder = defaultTargetBuilders[targetName];
   const targetConfig = project.architect && project.architect[targetName] ||
                        project.targets && project.targets[targetName];
-
-  if (!targetConfig || targetConfig['builder'] !== defaultBuilder) {
-    throw new SchematicsException(
-      `Your project is not using the default builders for "${targetName}". The Angular Material ` +
-      'schematics can only be used if the original builders from the Angular CLI are configured.');
+  const isDefaultBuilder = targetConfig && targetConfig['builder'] === defaultBuilder;
+
+  // Because the build setup for the Angular CLI can be customized by developers, we can't know
+  // where to put the theme file in the workspace configuration if custom builders are being
+  // used. In case the builder has been changed for the "build" target, we throw an error and
+  // exit because setting up a theme is a primary goal of `ng-add`. Otherwise if just the "test"
+  // builder has been changed, we warn because a theme is not mandatory for running tests
+  // with Material. See: https://github.com/angular/material2/issues/14176
+  if (!isDefaultBuilder && targetName === 'build') {
+    throw new Error(`Your project is not using the default builders for "${targetName}". The ` +
```

Comment:
Perhaps this should remain a SchematicsException?

```suggestion
    throw new SchematicsException(`Your project is not using the default builders for "${targetName}". The ` +
```

Reply:
Not too sure. I don't see big value in using `SchematicsException`. We mix these up a lot in our schematics, so I will make a note on making that consistent in the future.